src/app/InteractionModelEngine.cpp

```diff
@@ -105,7 +102,29 @@ CHIP_ERROR InteractionModelEngine::NewCommandSender(CommandSender ** const apCom
     return err;
 }
 
-void InteractionModelEngine::OnUnknownMsgType(Messaging::ExchangeContext * apEc, const PacketHeader & aPacketHeader,
+CHIP_ERROR InteractionModelEngine::NewReadClient(ReadClient ** const apReadClient, InteractionModelDelegate * apDelegate)
+{
+    CHIP_ERROR err = CHIP_ERROR_NO_MEMORY;
+
+    for (auto & readClient : mReadClients)
+    {
+        if (readClient.IsFree())
+        {
+            *apReadClient = &readClient;
+            err           = readClient.Init(mpExchangeMgr, apDelegate);
+            if (CHIP_NO_ERROR != err)
+            {
+                *apReadClient = nullptr;
+                return err;
+            }
+            break;
```

Comment:
should this be 'return err' in all cases (and the if branch just sets result to NULL)?

Reply:
Yep, done, thanks